glslc/src/file_includer.h

```diff
@@ -45,6 +49,7 @@ class FileIncluder : public shaderc::CompileOptions::IncluderInterface {
   shaderc_includer_response response_;
   std::vector<char> file_content_;
   std::string file_full_path_;
+  std::set<std::string> file_path_trace_;
```

Comment:
Current paths are not absolute, so in some cases we can not behave as gcc/clang. When input file name is represented in absolute path, gcc/clang always use absolute for the dependent file names in the .d files. If the input files are relative paths, gcc/clang uses relative paths too.

Because we don't have context aware includer yet, I think the handler can not support this function right now as the dependency info dumping handler just use the filenames "as is" and has no idea about files.

Currently, when we an input file which does not include other files, and is represented in absolute path, we can have its absolute path in the .d file, because the handler uses the input source file name "as is", and it is not from the record of includer.